### PR TITLE
Port System.Runtime.InteropServices new docs

### PR DESCRIPTION
**Pull request by carlossanlop**
@dotnet/interop-contrib @directhex @gewarren PTAL

### CODE REVIEW

File: xml/System.Runtime.InteropServices.Swift/SwiftIndirectResult.xml [hunk 1]
```diff
@@ -24,8 +24,14 @@
...
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Represents the Swift return buffer context.</summary>
+    <remarks>
+      <para> This struct is used to access the return buffer when interoping with Swift functions that return non-frozen structs.
+ It provides a pointer to the memory location where the result should be stored.
+ </para>
+      <para> Here's an example of how a SwiftIndirectResult can be declared:
+ <code lang="csharp"> [UnmanagedCallConv(CallConvs = [typeof(CallConvSwift)])] [LibraryImport("SwiftLibrary", EntryPoint = "export")] public static extern void swiftFunction(SwiftIndirectResult result); </code></para>
```
Comment (AaronRobinsonMSFT) [inline xml/System.Runtime.InteropServices.Swift/SwiftIndirectResult.xml:33]: With this code be pretty-printed? Having it as a single line makes for poor documentation.

**Review by tannergooding — APPROVED** [review]
NFloat LGTM

File: xml/System.Runtime.InteropServices.Swift/SwiftIndirectResult.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.Runtime.InteropServices.Swift/SwiftIndirectResult.xml:33]: Here's the preview, it looks icky: https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.swift.swiftindirectresult?view=net-9.0&branch=pr-en-us-10154#remarks

Let me find out how to make it look better.

Comment (carlossanlop) [reply xml/System.Runtime.InteropServices.Swift/SwiftIndirectResult.xml]: Fixed. Let's wait for the preview now.

Comment (carlossanlop) [reply xml/System.Runtime.InteropServices.Swift/SwiftIndirectResult.xml]: @AaronRobinsonMSFT looks nice now! https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.swift.swiftindirectresult?view=net-9.0&branch=pr-en-us-10154#examples

image

**Review by jkoritzinsky — APPROVED** [review]

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**